### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by algoliareadmebot**
Pull Request README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -721,9 +732,9 @@ They are three scopes:
 **Attributes**
 - [attributesToIndex](#attributestoindex) `settings`
 - [attributesForFaceting](#attributesforfaceting) `settings`
-- [attributesToRetrieve](#attributestoretrieve) `settings`
+- [attributesToRetrieve](#attributestoretrieve) `settings`, `search`
 - [unretrievableAttributes](#unretrievableattributes) `settings`
-- [attributesToRetrieve](#attributestoretrieve) `search`
+- [attributesToRetrieve](#attributestoretrieve) `settings`, `search`
```
Comment (aseure) [inline README.md]: `attributesToRetrieve` is duplicated.

File: README.md [hunk 2]
```diff
@@ -1257,7 +1308,9 @@ if you have several geo-locations in your record).
 #### queryType
 
 - scope: `settings`, `search`
-- default: `prefixLast`
+- type: `enum`
+- default: `<%= puts({'C#' => 'PREFIX_LAST', 'Java' => 'PREFIX_ALL', 'Android' => 'PREFIX_ALL'}, 'prefixLast') %>`
+
```
Comment (aseure) [inline README.md]: Code from the generator.

File: README.md [hunk 3]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
 
 
 
+
+
+## Synonyms
+
+### Save synonym - `saveSynonym`
```
Comment (aseure) [inline README.md]: `AddSynonym`

File: README.md [hunk 4]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
+In this example, we specify true to forward the creation to slave indices.
+By default the behavior is to save only on the specified index.
+
+uniqueID := "a-unique-identifier"
+synonym := algoliasearch.NewSynonym(uniqueID, []string{
+    "car", "vehicle", "auto",
+})
+res, err := index.AddSynonym(uniqueID, synonym, true)
+
+### Batch synonyms - `batchSynonyms`
```
Comment (aseure) [inline README.md]: `BatchSynonyms`

File: README.md [hunk 5]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
+Updating the value of a specific synonym record is the same as creating one.
+Make sure you specify the same objectID used to create the record and the synonyms
+will be updated.
+When updating multiple synonyms in a batch call (but not all synonyms),
+make sure you set replaceExistingSynonyms to false (or leave it out,
+false is the default value).
+Otherwise, the entire synonym list will be replaced only partially with the records
+in the batch update.
+
+### Delete Synonyms - `delete_synonyms`
```
Comment (aseure) [inline README.md]: `DeleteSynonym`

File: README.md [hunk 6]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
+
+Use the normal index delete method to delete synonyms,
+specifying the objectID of the synonym record you want to delete.
+Forward the deletion to slave indices by setting the forwardToSlaves parameter to true.
+
+// Delete and forward to slaves
+res, err := index.DeleteSynonym("a-unique-identifier", true)
+
+### Clear all synonyms - `clearSynonyms`
```
Comment (aseure) [inline README.md]: `ClearSynonyms`

File: README.md [hunk 7]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
+there would be a short period of time during which the index would have no synonyms
+at all.
+
+To atomically replace all synonyms of an index,
+use the batch method with the replaceExistingSynonyms parameter set to true.
+
+// Clear synonyms and forward to slaves
+res, err := index.ClearSynonyms(true)
+
+### Get synonym - `getSynonym`
```
Comment (aseure) [inline README.md]: `GetSynonym`

File: README.md [hunk 8]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
+
+Search for synonym records by their objectID or by the text they contain.
+Both methods are covered here.
+
+synonym, err := index.GetSynonym("a-unique-identifier")
+
+### Search synonyms - `searchSynonyms`
```
Comment (aseure) [inline README.md]: `SearchSynonym`

File: README.md [hunk 9]
```diff
@@ -1697,16 +1773,105 @@ index.search('another query', function(err, content) {
...
 ### Custom batch - `Batch`
 
 You may want to perform multiple operations with one API call to reduce latency.
 We expose four methods to perform batch operations:
- * `AddObjects`: Add an array of objects using automatic `objectID` assignment.
- * `UpdateObjects`: Add or update an array of objects that contains an `objectID` attribute.
- * `DeleteObjects`: Delete an array of objectIDs.
- * `PartialUpdateObjects`: Partially update an array of objects that contain an `objectID` attribute (only specified attributes will be updated).
+ * Add objects - `AddObjects`: Add an array of objects using automatic `objectID` assignment.
+ * Update objects - `SaveObjects`: Add or update an array of objects that contains an `objectID` attribute.
```
Comment (aseure) [inline README.md]: `UpdateObjects`

**Comment by redox** [comment]
@aseure I think you'll need to do the same kind of PR I did https://github.com/algolia/algoliasearch-client-readme-generator/pull/98

File: README.md [hunk 10]
```diff
@@ -2036,7 +2201,7 @@ res, err := client.DeleteKey("f420238212c54dcfad07ea0aa6d5c45f")
 res, err := index.DeleteKey("71671c38001bf3ac857bc82052485107")
 ```
 
-### Get key permissions - `GetUserKeyACL`
+### Get key permissions - `getUserKeyACL`
```
Comment (aseure) [inline README.md]: `GetUserKey`

File: README.md [hunk 11]
```diff
@@ -2049,7 +2214,7 @@ key, err := client.GetKey("f420238212c54dcfad07ea0aa6d5c45f")
 key, err = index.GetKey("71671c38001bf3ac857bc82052485107")
 ```
 
-### Multiple queries - `MultipleQueries`
+### Multiple queries - `multipleQueries`
```
Comment (aseure) [inline README.md]: `MultipleQueries`

File: README.md [hunk 12]
```diff
@@ -2083,7 +2248,7 @@ You can specify a `strategy` parameter to optimize your multiple queries:
 
 
 
-### Get Logs - `GetLogs`
+### Get Logs - `getLogs`
```
Comment (aseure) [inline README.md]: `GetLogs`

**Comment by aseure** [comment]
@redox Yes indeed. First, I was just listing all the mistakes.

File: README.md [hunk 13]
```diff
@@ -641,13 +647,15 @@ res, err := index.SetSettings(settings)
 
 <!--PARAMETERS_LINK-->
 
-Here is the list of parameters you can use with the set settings method (`indexing` [scope](#scope)):
+Here is the list of parameters you can use with the set settings method (`indexing` [scope](#scope))
+
+
 Parameters that can be override at search time also have the `indexing` [scope](#scope)
```
Comment (aseure) [inline README.md]: Is it expected @maxiloc?

Comment (maxiloc) [inline README.md]: no there is a trailing newline on the generator but it's not an issue

**Pull request merged**